src/inference/dev_api/threading/ie_istreams_executor.hpp

```diff
@@ -95,6 +97,12 @@ class INFERENCE_ENGINE_API_CLASS(IStreamsExecutor) : public ITaskExecutor {
                                        //!< starting from offset
         int _threads = 0;              //!< Number of threads distributed between streams.
                                        //!< Reserved. Should not be used.
+        mutable int _big_core_streams = 0;
+        mutable int _small_core_streams = 0;
+        mutable int _threads_per_stream_big = 0;
+        mutable int _threads_per_stream_small = 0;
+        mutable int _small_core_offset = 0;
+        enum StreamMode { DEFAULT, AGGRESSIVE, LESSAGGRESSIVE };
```

Comment:
In general this enum for streams logic configuration seems to be non-scalable. I would prefer not to add MUCHLESSAGRESSIVE mode in future :) Moreover internal logic for ARM processors might be different, so not sure this is good solution.
Maybe more scalable way is to provide prefered_cores_per_stream value as a parameter?

Reply:
Agree with you on scalable approach. 

 Current implementation is aligned with exist CORES implementation.

Reply:
Will follow up in next PR